v2/apiserver/schemas/project.json

```diff
@@ -223,6 +223,14 @@
 						}
 					],
 					"description": "The URL for cloning a git project"
+        },
+				"commit": {
+					"type": "string",
+					"description": "A git commit sha"
```

Comment:
Can we restrict this field to hexadecimal characters only?

`event.json` probably deserves the same change... and as you noted, common types can probably be consolidated into some kind of common / library file.

Reply:
Updated.  Could you check my work on the pattern?  Also, I added a min and max length according to git short/full sha conventions -- overkill?

Reply:
I think that's reasonable.